packages/bot-web-ui/src/utils/__tests__/journal-notifications.spec.tsx

```diff
@@ -0,0 +1,116 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render } from '@testing-library/react';
+import { messageWithButton } from 'Components/notify-item';
+import { isCustomJournalMessage } from '../journal-notifications';
+import { localize } from '@deriv/translations';
+
+describe('isCustomJournalMessage', () => {
+    const centerAndHighlightBlock = jest.fn();
+    const showErrorMessage = jest.fn(() =>
+        render(
+            messageWithButton({
+                unique_id: 1,
+                type: 'error',
+                message: '',
+                btn_text: localize('Go to block'),
+                onClick: () => {
+                    centerAndHighlightBlock();
+                },
+            })
+        )
+    );
+
+    const pushMessage = jest.fn();
+
+    beforeEach(() => {
+        showErrorMessage.mockClear();
+        centerAndHighlightBlock.mockClear();
+        pushMessage.mockClear();
+    });
+
+    it('should show error message with button when message is undefined and variable_name is not null', () => {
+        const message = {
+            message: undefined,
+            block_id: 123,
+            variable_name: 'testVariable',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        expect(showErrorMessage).toHaveBeenCalledWith(expect.any(Object));
+    });
+
+    it('should show error message with button when message is not undefined and variable_name is null', () => {
+        const message = {
+            message: [['item1']],
+            block_id: 123,
+            variable_name: 'null',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        setTimeout(() => {
```

Comment:
@maryia-matskevich-deriv can we remove this setTimeout too ?

Reply:
done